pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -15,7 +15,29 @@ rec {
         --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
     '';
 
-    phases = "unpackPhase installPhase";
+    libnativeFixup =
+      let
+        arch =
+          { "x86_64-linux" = "linux-amd64";
+            "i686-linux" = "linux-i386";
+          }."${stdenv.system}" or "";
+       in ''
+        mkdir patching
+        pushd patching
+        jar xf $out/lib/gradle/lib/native-platform-${arch}-0.10.jar
+        ${patchelf}/bin/patchelf --set-rpath "${stdenv.cc.cc}/lib:${stdenv.cc.cc}/lib64" net/rubygrapefruit/platform/${arch}/libnative-platform.so
+        jar cf native-platform-${arch}-0.10.jar .
+        mv native-platform-${arch}-0.10.jar $out/lib/gradle/lib/
+        popd
+
+        # The scanner doesn't pick up the runtime dependency in the jar.
+        # Manually add a reference where it will be found.
+        mkdir $out/nix-support
+        echo ${stdenv.cc.cc} > $out/nix-support/manual-runtime-dependencies
+      '';
+
+    fixupPhase = if stdenv.isLinux then libnativeFixup else "true";
```

Comment:
More nitpicks: I think this is cleaner as
```
fixupPhase = if (!stdenv.isLinux) then ":" else ''
   foo
   bar
''
```
assuming this will always be linux only. You could also try `optionalString stdenv.isLinux`. You could then also replace the `arch` lookup with simply `stdenv.is64bit` (well, I guess you could do that anyway, but now it would be even clearer that the fixup code is linux only to begin with, so there'd be no need to restate that assumption).